components/playlist/resources/api/api.ts

```diff
@@ -47,9 +48,14 @@ class API {
         playlistId, mojoUrl)
   }
 
-  addMediaFilesFromOpenTabsToPlaylist (playlistId: string) {
-    this.#pageHandler.addMediaFilesFromOpenTabsToPlaylist(
-        playlistId)
+  addMediaFilesFromActiveTabToPlaylist (playlistId: string) {
+    PlaylistMojo.PageHandlerFactory.getRemote().getActiveTabId().then(({ windowId, tabId }) => {
```

Comment:
this has exactly the same problem as the original version in terms of async calls. I was looking at upstream examples and `BookmarksPageHandler::BookmarkCurrentTab` and `ReadingListPageHandler::AddCurrentTab` both use `Browser* browser = chrome::FindLastActive();` so I guess that's fine here after all.

Reply:
Reverted.